dotCMS/src/main/java/com/dotcms/vanityurl/model/CachedVanityUrl.java

```diff
@@ -17,141 +24,66 @@
  */
 public class CachedVanityUrl implements Serializable {
 
-    private static final long serialVersionUID = 1L;
-    private final Pattern pattern;
-    private final String vanityUrlId;
-    private final String url;
-    private final String siteId;
-    private final long languageId;
-    private final String forwardTo;
-    private final int response;
-    private final int order;
+    static final long serialVersionUID = 1L;
+    final public Pattern pattern;
+    final public String vanityUrlId;
+    final public String url;
+    final public String siteId;
+    final public long languageId;
+    final public String forwardTo;
+    final public int response;
+    final public int order;
 
     /**
      * Generate a cached Vanity URL object
      *
      * @param vanityUrl The vanityurl Url to cache
      */
+
     public CachedVanityUrl(final VanityUrl vanityUrl) {
-        //if the VanityUrl URI is not a valid regex
-        this.vanityUrlId = vanityUrl.getIdentifier();
-        final String regex = normalize(vanityUrl.getURI(), CACHE_404_VANITY_URL.equals(this.vanityUrlId));
-        this.pattern = Pattern.compile(regex, Pattern.CASE_INSENSITIVE);
-        this.url = vanityUrl.getURI();
-        this.languageId = vanityUrl.getLanguageId();
-        this.siteId = vanityUrl.getSite();
-        this.forwardTo = vanityUrl.getForwardTo();
-        this.response = vanityUrl.getAction();
-        this.order    = vanityUrl.getOrder();
+        this(vanityUrl.getIdentifier(),vanityUrl.getURI(),vanityUrl.getLanguageId(),vanityUrl.getSite(),vanityUrl.getForwardTo(),vanityUrl.getAction());
     }
 
-    /**
-     * Generates a CachedVanityUrl from another given CachedVanityUrl
-     *
-     * @param fromCachedVanityUrl VanityURL to copy
-     * @param url url to override in the created copy
-     */
-    public CachedVanityUrl(CachedVanityUrl fromCachedVanityUrl, String url) {
-
-        this.vanityUrlId = fromCachedVanityUrl.getVanityUrlId();
-        //if the VanityUrl URI is not a valid regex
-        final String regex = normalize(url, CACHE_404_VANITY_URL.equals(this.vanityUrlId));
+    public CachedVanityUrl(String vanityUrlId,String url, long languageId, String siteId,String forwardTo,int response) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'vanityUrlId' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)